ff_espn_api/box_player.py

```diff
@@ -1,19 +1,35 @@
-from .constant import POSITION_MAP
+from .constant import POSITION_MAP, PRO_TEAM_MAP
 from .player import Player
+from datetime import datetime, timedelta
 
 class BoxPlayer(Player):
     '''player with extra data from a matchup'''
-    def __init__(self, data):
+    def __init__(self, data, pro_schedule, positional_rankings, week):
         super(BoxPlayer, self).__init__(data)
-        self.slot_position = POSITION_MAP[data['lineupSlotId']]
+        self.slot_position = 'FA'
         self.points = 0
         self.projected_points = 0
+        self.pro_opponent = "None" # professional team playing against
+        self.pro_pos_rank = 0 # rank of professional team against player position
+        self.game_played = 100 # 0-100 for percent of game played
 
-        player_stats = data['playerPoolEntry']['player']['stats']
+        if 'lineupSlotId' in data:
+            self.slot_position = POSITION_MAP[data['lineupSlotId']]
+
+        player = data['playerPoolEntry']['player'] if 'playerPoolEntry' in data else data['player']

```

Comment:
Because this gets player team data at the current time, opponent information in incorrect for players who have since been traded.